vck/src/commonMain/kotlin/at/asitplus/wallet/lib/iso/IssuerSignedItemSerializer.kt

```diff
@@ -92,16 +95,26 @@ object IssuerSignedItemSerializer : KSerializer<IssuerSignedItem> {
         )
     }
 
-    private fun CompositeDecoder.decodeAnything(index: Int): Any {
-        runCatching { return decodeStringElement(descriptor, index) }
-        runCatching { return decodeSerializableElement(descriptor, index, ByteArraySerializer()) }
-        runCatching { return decodeBooleanElement(descriptor, index) }
-        runCatching { return decodeSerializableElement(descriptor, index, LocalDate.serializer()) }
-        runCatching { return decodeSerializableElement(descriptor, index, InstantStringSerializer()) }
+    private fun CompositeDecoder.decodeAnything(index: Int, elementIdentifier: String): Any {
+        if (namespace.isBlank()) Napier.w { "This decoder is not namespace-aware! Unspeakable things may happen…" }
+
+        //Tags are not read out here but skipped because `decodeElementIndex` is never called, so we cannot discriminate
+        //technically, this should be a good thing though, because otherwise we'd consume more from the input
         runCatching {
-            return CborCredentialSerializer.decode(descriptor, index, this)
-                ?: throw IllegalArgumentException("Could not decode value at $index")
+
+            CborCredentialSerializer.decode(descriptor, index, this, elementIdentifier, namespace)?.let {
+                return it
+            }
+                ?: Napier.w { "Could not find a registered decoder for namespace $namespace and elementIdentifier $elementIdentifier. Falling back to defaults" }
+
         }
+
+        //These are the ones that map to different CBOR data types, the rest don't, so if it is not registered, we'll lose type information
+        runCatching { return decodeStringElement(descriptor, index) }.exceptionOrNull()?.printStackTrace()
```

Comment:
Do we really want to print the stack trace everytime?

Reply:
nope. removed. thanks!